### PR TITLE
Add resources for documentation

### PR DESCRIPTION
**Pull request by xgarrido**
This PR adds resources for a proper documentation with https://readthedocs.org/projects/lat-mflike

- the original notebook has been split into several smaller tutorials available in `docs/source/notebooks` directory
- the `mflike` loading is done in the first tutorial; other tutorials rely on the first one to load `mflike` with the same setup (this prevent copy/paste multiple times the same code)
- part of the original tutorial has been modified (such as the use of `sacc` utilities functions that has been removed) in order to do things as much as possible with the `mflike` object
- since it's a tutorial, `camb` is used with minimal settings (*i.e.* `lens_potential_accuracy=1`). Higher accuracy settings are proposed and user can easily switch between both

### CODE REVIEW

**Comment by xgarrido** [comment]
The readthedoc document can be see [here](https://lat-mflike.readthedocs.io/en/feature-docs)

File: docs/source/notebooks/tutorial_residuals.ipynb [hunk 1]
```diff
@@ -0,0 +1,1067 @@
...
+   "id": "d17b0825-d1ae-46b5-a8b3-6941f0355a9b",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "Simulations and data as well as covariance are stored as `sacc` objects. For more details regarding `sacc` format, you can have a look to the tutorial [notebooks](https://github.com/simonsobs/sacc/tree/master/examples) or you can refer to the official [documentation](https://sacc.readthedocs.io/en/latest/). Here we will read the power spectra for all the simulations and compare its mean value to theory + foregrounds (without and with systematics). \n",
```
Comment (sgiardie) [inline docs/source/notebooks/tutorial_residuals.ipynb:51]: The link to [the sacc notebook tutorial](https://github.com/simonsobs/sacc/tree/master/examples) does not work for me

**Comment by xgarrido** [comment]
Thanks for the dead link.

This [documentation](https://soliket.readthedocs.io/en/latest/mflike.html) is pretty cool. Can we backport to `mflike` ? I guess we have to add docstrings to the code, do you want me to do it or would prefer to do it by yourself (I think you did for `soliket`)

**Comment by sgiardie** [comment]
Sure, I can work on the docstrings!

**Comment by xgarrido** [comment]
I have added `mflike.rst` and `theoryforge.rst` files that will get the docstrings from both classes.

**Comment by xgarrido** [comment]
Thanks a lot @sgiardie. I did fix some issue with docstrings such as escaping `\**params` but I think we now have a pretty nice documentation for `mflike`

**Comment by xgarrido** [comment]
I'm going to move the API section after the tutorials (as you initially wanted, it's better like that)

**Comment by sgiardie** [comment]
looks good to me!

**Pull request merged**